dali/ft/filecopy.cpp

```diff
@@ -592,6 +596,24 @@ FileSprayer::FileSprayer(IPropertyTree * _options, IPropertyTree * _progress, IR
     encryptKey.set(options->queryProp(ANencryptKey));
     decryptKey.set(options->queryProp(ANdecryptKey));
 
+    fileUmask = -1;
+    if (options->hasProp(ANumask))
+    {
+        StringBuffer umaskStr;
+        options->getProp(ANumask, umaskStr);
+        errno = 0;
+        fileUmask = (int)strtol(umaskStr.str(), NULL, 8);
+        if (errno)
```

Comment:
strtol does not set errno unless value overflows, AFAIK - it is not set for parsing errors.